test/jdk/javax/swing/JButton/TimeChangeButtonClickTest.java

```diff
@@ -26,195 +26,103 @@
-
-        if (!status) {
-            System.out.println("Test timed out.");
-        }
-
-        SwingUtilities.invokeAndWait(new Runnable() {
-            @Override
-            public void run() {
-                try {
-                    test.disposeUI();
-                } catch (Exception ex) {
-                    throw new RuntimeException("Exception while disposing test UI");
-                }
-            }
-        });
-
-        if (test.testResult == false) {
-            throw new RuntimeException("Test Failed.");
-        }
-    }
-}
-
-class TestUI {
-
-    private static JFrame mainFrame;
-    private static JPanel mainControlPanel;
-
-    private static JTextArea instructionTextArea;
-
-    private static JPanel resultButtonPanel;
-    private static JButton passButton;
-    private static JButton failButton;
-
-    private static JPanel testPanel;
-    private static JButton testButton;
-    private static JLabel buttonPressCountLabel;
-
-    private static GridBagLayout layout;
-    private final CountDownLatch latch;
-    public boolean testResult = false;
-    private int buttonPressCount = 0;
-
-    public TestUI(CountDownLatch latch) throws Exception {
-        this.latch = latch;
+    private static final String INSTRUCTIONS = """
+            <html><body>
+            <ol>
+            <li>
+                <strong>Test 1:</strong>
+                <ol style="margin-top: 0px">
+                    <li>Click <b>Test Button</b> with left mouse button</li>
+                    <li>Observe: Button Press count change to 1</li>
+                </ol>
+            </li>
+            <li>
+                <strong>Test 2:</strong>
+                <ol style="margin-top: 0px">
+                    <li>Change the system time to one hour less than current time</li>
+                    <li>Click <b>Test Button</b> with left mouse button</li>
+                    <li>Observe: Button Press count change to 2</li>
+                </ol>
+            </li>
+            <li>
+                <strong>Test 3:</strong>
+                <ol style="margin-top: 0px">
+                    <li>Change the system time by adding two hours</li>
+                    <li>Click <b>Test Button</b> with left mouse button</li>
+                    <li>Observe: Button Press count changes to 3</li>
+                </ol>
+            </li>
+
+            <li style="margin-top: 8px; padding-top: 8px; border-top: 1px solid">
+            Restore the system time.</li>
+            <li style="margin-top: 8px; padding-top: 8px; border-top: 1px solid">
+            Press <b>Pass</b> if Button Press count is 3,
+            and <b>Fail</b> otherwise.</li>
+            </ol>
+            </body></html>
+            """;
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame.builder()
+                      .instructions(INSTRUCTIONS)
+                      .rows(20)
+                      .columns(40)
```

Comment:
```suggestion
                      .columns(40)
                      .testTimeOut(10)
```

We do have a pause timer option but I think a timeout of 8-10 mins is better than the default of 5 mins since it allows more time for the user to change the time settings.

Reply:
Actually, I think what is happening is that when the time is set forward an hour (+2 hours forward from 1 hour behind as in instructions) the test will automatically time out because it thinks an hour has passed. I thought initially it was me being slow on changing the time settings but even if you do everything quickly the test always fails exactly when you set the time forward.

So I think with the way the timeout is implemented the test is broken with PassFailJFrame.

Reply:
What an .. interesting .. test. No idea what this does to the test harness too.
FWIW I think this test should be deleted rather than trying to make it nicer.

Reply:
> Actually, I think what is happening is that when the time is set forward an hour (+2 hours forward from 1 hour behind as in instructions) the test will automatically time out because it thinks an hour has passed. I thought initially it was me being slow on changing the time settings but even if you do everything quickly the test always fails exactly when you set the time forward.

Good catch. I hadn't noticed PFJ timeout.

Reply:
> The workaround could be to pause the timeout. In addition to that, we could set the timeout to 15 minutes. If the user changes time not -1 / +2 hours but rather -5 / +10 minutes, it should be enough.

Either approach sounds good. Easy workaround is to include pause timer in the test instructions.

```
Pause the timer and change the system time to one hour less than current time.
......
.....
Pause the timer and change the system time by adding two hours
```

Yes I agree, retaining the test might be a good idea since it was created for -
[JDK-7096375](https://bugs.openjdk.org/browse/JDK-7096375).

Reply:
Sounds good for a workaround.